lambda/handler.go

```diff
@@ -23,6 +23,8 @@ type handlerOptions struct {
 	jsonResponseEscapeHTML   bool
 	jsonResponseIndentPrefix string
 	jsonResponseIndentValue  string
+	enableSIGTERM            bool
+	sigtermCallbacks         []func()
```

Comment:
Do we need an excplicit `enableSIGTERM` flag + another for registering callbacks? Would users ever enable SIGTERM without configuring handlers, or is `enableSIGTERM` redundant?

If not, I suggest combining into a single configuration option eg. `registerCallbacksSIGTERM`

Reply:
Yeah the signal handlers can be setup anywhere within the program, including in libraries outside of the function author's control. I include an example of this in the test cases. For these passing additional callbacks into `WithEnableSIGTERM` is optional.

```
func init() {
 	// conventional SIGTERM callback
 	signaled := make(chan os.Signal, 1)
 	signal.Notify(signaled, syscall.SIGTERM)
 	go func() {
 		<-signaled
 		fmt.Println("I've been TERMINATED!")
 	}()

 }
```

Reply:
Understood, makes sense - thanks!